src/test/reporters/raw.ts

```diff
@@ -0,0 +1,244 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { FullProject } from '../../../types/test';
+import { FullConfig, Location, Suite, TestCase, TestError, TestResult, TestStatus, TestStep } from '../../../types/testReporter';
+import { assert, calculateSha1 } from '../../utils/utils';
+import { sanitizeForFilePath } from '../util';
+import { serializePatterns, toPosixPath } from './json';
+
+export type JsonStats = { expected: number, unexpected: number, flaky: number, skipped: number };
+export type JsonLocation = Location;
+export type JsonStackFrame = { file: string, line: number, column: number };
+
+export type JsonReport = {
+  config: JsonConfig,
+  project: JsonProject,
+  suites: JsonSuite[],
+};
+
+export type JsonConfig = Omit<FullConfig, 'projects'>;
+
+export type JsonProject = {
+  metadata: any,
+  name: string,
+  outputDir: string,
+  repeatEach: number,
+  retries: number,
+  testDir: string,
+  testIgnore: string[],
+  testMatch: string[],
+  timeout: number,
+};
+
+export type JsonSuite = {
+  title: string;
+  location?: JsonLocation;
+  suites: JsonSuite[];
+  tests: JsonTestCase[];
+};
+
+export type JsonTestCase = {
+  testId: string;
+  title: string;
+  location: JsonLocation;
+  expectedStatus: TestStatus;
+  timeout: number;
+  annotations: { type: string, description?: string }[];
+  retries: number;
+  results: JsonTestResult[];
+  ok: boolean;
+  outcome: 'skipped' | 'expected' | 'unexpected' | 'flaky';
+};
+
+export type TestAttachment = {
+  name: string;
+  path?: string;
+  body?: Buffer;
+  contentType: string;
+};
+
+export type JsonAttachment = {
+  name: string;
+  path: string;
+  contentType: string;
+};
+
+export type JsonTestResult = {
+  retry: number;
+  workerIndex: number;
+  startTime: string;
+  duration: number;
+  status: TestStatus;
+  error?: TestError;
+  attachments: JsonAttachment[];
+  steps: JsonTestStep[];
+};
+
+export type JsonTestStep = {
+  title: string;
+  category: string,
+  startTime: string;
+  duration: number;
+  error?: TestError;
+  steps: JsonTestStep[];
+  log?: string[];
+};
+
+class RawReporter {
+  private config!: FullConfig;
+  private suite!: Suite;
+
+  onBegin(config: FullConfig, suite: Suite) {
+    this.config = config;
+    this.suite = suite;
+  }
+
+  async onEnd() {
+    const projectSuites = this.suite.suites;
+    for (const suite of projectSuites) {
+      const project = (suite as any)._projectConfig as FullProject;
+      assert(project, 'Internal Error: Invalid project structure');
+      const reportFolder = path.join(project.outputDir, 'report');
+      fs.mkdirSync(reportFolder, { recursive: true });
+      let reportFile: string | undefined;
+      for (let i = 0; i < 10; ++i) {
+        reportFile = path.join(reportFolder, sanitizeForFilePath(project.name || 'project') + (i ? '-' + i : '') + '.report');
```

Comment:
Maybe keep '.json'. for tooling, e.g. opening in IDE?

Reply:
I wouldn't want it to be opened in vscode. Will probably use cbor to defend myself :)